lib/debug: throw functions that have been deprecated since 2018

Artturin opened the pull request:
traceShowVal
traceShowValMarked
attrNamesToStr
showVal
traceXMLVal
traceXMLValMarked
traceCall
traceCall2
traceCall3
traceValIfNot
addErrorContextToAttrs
traceCallXml

they were deprecated in https://github.com/NixOS/nixpkgs/pull/38368

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

infinisil reviewed (changes requested):
To deprecate something, it first needs an explanatory warning in one release, an explanatory error in another, and then it can be removed in a further release down the line.

This means for functions without a warning you should add one and for ones with a warning you should make it an error instead.

Artturin committed:
lib/debug: throw functions that have been deprecated since 2018

traceShowVal
traceShowValMarked
attrNamesToStr
showVal
traceXMLVal
traceXMLValMarked
traceCall
traceCall2
traceCall3
traceValIfNot
addErrorContextToAttrs
traceCallXml

they were deprecated in https://github.com/NixOS/nixpkgs/pull/38368

Artturin commented:
> To deprecate something, it first needs an explanatory warning in one release, an explanatory error in another, and then it can be removed in a further release down the line.
> 
> This means for functions without a warning you should add one and for ones with a warning you should make it an error instead.

i've added throws and `FILL IN`'s for functions which i don't know how they were used and don't know the replacement

infinisil commented on lib/debug.nix:
```diff
@@ -206,86 +206,63 @@ rec {
 
-  traceShowVal = x: trace (showVal x) x;
-  traceShowValMarked = str: x: trace (str + showVal x) x;
+  traceShowVal =
+    throw ( "`traceShowVal` is deprecated "
+          + "and will be removed in release 23.05. "
+          + "Please use `FILL IN`." );
```
Again, if there was no warning already, it should become a warning, not an error.

Artturin replied:
These used showVal which had a warning already so imo its good enough.

infinisil replied:
Ahhh! Sorry for that then, makes sense

infinisil reviewed (changes requested):
To figure out why functions are deprecated you need to dig through `git blame` history. In this case I could find https://github.com/NixOS/nixpkgs/pull/38368 which contains some explanations. Pinging @Profpatsch for this also seems appropriate
